### PR TITLE
Instance route in admin panel needs improvement

### PR DESCRIPTION
**Pull request by kimenyikevin**
## Summary

A summary of changes being made in this PR

## Checklist
- [x] Pre-push checks pass `npm run check`
  - [x] Linter passing via `npm run lint`
  - [ ] Unit & Integration tests passing via `npm run test:packages`
  - [x] Docker build process passing via `npm run build-client`
- [x] If this PR is still a WIP, convert to a draft 
- [x] When this PR is ready, mark it as "Ready for review"
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewers

## References

References to pertaining issue(s)

## QA Steps

1. `git checkout pr_branch_name`
2. `npm install`
3. `npm run dev-reinit`
4. `npm run dev`

List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos.

## Reviewers

@HexaField @speigg @NateTheGreatt

#4611

## Task to be completed 
- Search instance on admin location 
- delete instance 
- define re-usable component 
- Merge all style in one file

### CODE REVIEW

File: packages/client-core/src/admin/components/Instance/InstanceTable.tsx [hunk 1]
```diff
@@ -88,9 +96,16 @@ const InstanceTable = (props: Props) => {
       channelId,
       action: (
         <>
-          <a href="#h" className={classes.actionStyle}>
-            {' '}
-            <span className={classes.spanDange}>Delete</span>{' '}
+          <a
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/components/Instance/InstanceTable.tsx:99]: do we need the `<></>` ? there is only a single element in actions

File: packages/client-core/src/admin/components/Instance/InstanceTable.tsx [hunk 2]
```diff
@@ -87,12 +95,17 @@ const InstanceTable = (props: Props) => {
...
-            {' '}
-            <span className={classes.spanDange}>Delete</span>{' '}
-          </a>
-        </>
+        <a
+          href="#h"
+          className={classes.actionStyle}
+          onClick={() => {
+            setPopConfirmOpen(true)
+            setInstanceId(id)
```
Comment (zulqarnainhanif) [inline packages/client-core/src/admin/components/Instance/InstanceTable.tsx:103]: getting error on deleting an instance

Screenshot 2022-01-13 at 4 48 31 PM

File: packages/server-core/src/networking/instance/instance.class.ts [hunk 3]
```diff
@@ -33,16 +33,7 @@ export class Instance extends Service {
...
-      if (search) {
-        q = {
-          [Op.or]: [
-            Sequelize.where(Sequelize.fn('lower', Sequelize.col('ipAddress')), {
-              [Op.like]: '%' + search.toLowerCase() + '%'
-            })
-          ]
-        }
-      }
+      let q = search ? { ended: false, ipAddress: { [Op.like]: `%${search}%` } } : { ended: false }
```
Comment (zulqarnainhanif) [inline packages/server-core/src/networking/instance/instance.class.ts:36]: search should work on Location column as well atleast

File: packages/client-core/src/admin/components/Instance/InstanceTable.tsx (hunk 2 above)
Comment (kimenyikevin) [reply packages/client-core/src/admin/components/Instance/InstanceTable.tsx:103]: I am not able to get this error. which step are you following to delete instance ?

Comment (zulqarnainhanif) [reply packages/client-core/src/admin/components/Instance/InstanceTable.tsx:103]: simply pressed delete and confirmed from popup and that instance had 0 users.

Comment (kimenyikevin) [reply packages/client-core/src/admin/components/Instance/InstanceTable.tsx:103]: I tried to test it, but I am not able to get an error. could you please test again with latest changes

Comment (zulqarnainhanif) [reply packages/client-core/src/admin/components/Instance/InstanceTable.tsx:103]: - open two instance of location/test in separate browsers
- then go to /admin/instance route
- You will see instances like in image#1
- delete instance with 3031 port image#2
- confirm the deletion and the screen will go blank like in image#3

image#1
Screenshot 2022-01-13 at 8 55 40 PM

image#2
Screenshot 2022-01-13 at 8 55 57 PM

image#3
Screenshot 2022-01-13 at 8 56 09 PM

**Review by zulqarnainhanif — APPROVED** [review]

File: packages/client-core/src/admin/services/InstanceService.ts [hunk 4]
```diff
@@ -52,10 +48,10 @@ export const useInstanceState = () => useState(state) as any as typeof state
 
 //Service
 export const InstanceService = {
-  fetchAdminInstances: async (incDec?: 'increment' | 'decrement', search: string = '') => {
+  fetchAdminInstances: async (incDec?: 'increment' | 'decrement', search: string | null = null) => {
     const dispatch = useDispatch()
     {
-      const skip = accessInstanceState().skip.value
+      const skip = 0
```
Comment (barankyle) [inline packages/client-core/src/admin/services/InstanceService.ts:54]: Aren't we going to need skip to change in order to paginate?

Comment (kimenyikevin) [reply packages/client-core/src/admin/services/InstanceService.ts:54]: Let me change it back, I was just testing

**Pull request merged**